RationaleMCP/0035/Readme.md

```diff
@@ -0,0 +1,129 @@
+
+# Contributor License Agreement
+All authors of this MCP or their organizations have signed the "Modelica Contributor License Agreement". 
+
+# Rationale
+Why should this feature be included? 
+* A better understanding of the libraries by users with different language background will make the libraries more attractive. 
+* Support cases might occur less often, due to better understanding of the description. 
+* A new translation is additionally a review of the current state and can result in a better quality of both language versions.
+* Users can even write their own translation, when having a template *.pot-File for the library. 
+
+The use case for the proposal is the MSL 4.0. 
+* POT-File of current MSL 4.0 library --> this translation template is provided and filled for the Hydraulics Pump. 
+(*add pictures*)
+
+Proposed changes to the specification are described in SpecificationText.md. It consists of a new section **13.6 Multilingual Descriptions** which contains:
+* Where the translation files to be stored
+* How does a GNU gettext translation template for a library look like and which kind of descriptive texts shall be included?
+* Which strings should be translated:
+
+The precise updated text of the specification is part of this branch/pull-request: Yes.
+
+# Backwards Compatibility
+The proposal is backwards compatible. All multilingual text changes can be applied without effect at existing Modelica code. 
+After the introduction of translation files for libraries, this files need to be adjusted with every library change which adds new commented parts or changes existing comments.
+
+# Tool Implementation
+
+## Experience with Prototype
+### Implementation steps
+#### 1. Read out a translation template file
+Create at first a translation template file \*.pot for a Modelica library. Means a (tool dependent) readout of all needed text strings into a *LibraryName*.pot-File. It contains: 
+- a header which concludes all necessary information
+![Header of a *.pot file](POT_Header.png)
+
+- for all original strings entries with following structure: 
+```
+    msgctxt "Hydraulics.Basics.TankOverflow"
+    msgid "Volume Flow Port A"
+    msgstr ""
+```
+
+Where *msgctxt* contains the name of the Modelica Class and *msgid* contains the textstring which is situated in this name space and shall be translated. Please regard that if a *msgid* string (in this example "Volume Flow Port A") is given more than once in a namespace, all occurrences are translated with the same translation!
+
+It shall be mandatory to use for *msgctxt* the full name of a class:
+- so a specific translation depending on the content in this namespace is possible, as every textstring may differ depending on the context,
+- using the class makes it easier for the Modelica tool to copy and re-arrange models and packages without loosing the already existing language information.
+
+The following Modelica constructs shall be read out / translated: 
+* syntactical comments

```

Comment:
Just to clarify, what is meant by the syntactical comments here?

Reply:
It refers to _description-string_ in the grammar.

Reply:
Then it's probably worth renaming it. It is probably left from the time when *description-string* was called *string_comment*.

Reply:
Yes, and it is now corrected.